aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3289 @@
+It is the Thumb equivalent of R_ARM_CALL and the same rules on conversion
+apply.  Bits 0-10 of the first half-word encode the most significant bits of
+the branch offset, bits 0-10 of the second half-word encode the least
+significant bits and the offset is in units of half-words.  Thus 22 bits encode
+a branch offset of ± 2\ :sup:`22` bytes.  When linking Armv6 (and later, see [ARM
+ARM]) Thumb code the range of the branch is increased by 2 bits, increasing the
+offset range to ± 2\ :sup:`24` bytes. The same relocation is used for both cases since
+a linker need only know that the code will run on a Thumb-2 (Armv6 and later)
+capable processor to exploit the additional range.
+
+The addend for B.W and B<cond>.W is the signed immediate quantity encoded in
+the instruction, extracted in a similar way to BL; for details see [ARMARM_].
+
+The conditions under which call and jump relocations are permitted to generate
+an ip-corrupting intra-call veneer,  and their behaviour in conjunction with
+unresolved weak references, are specified in `Static Arm relocations`_
+under the heading Call and Jump relocations.
+
+Static miscellaneous relocations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+R_ARM_NONE records that the section containing the place to be relocated
+depends on the section defining the symbol mentioned in the relocation
+directive in a way otherwise invisible to the static linker.  The effect is to
+prevent removal of sections that might otherwise appear to be unused.
+
+R_ARM_V4BX records the location of an Armv4t BX instruction.  This enables a
+static linker to generate Armv4 compatible images from Armv4t objects
+containing only Arm code by converting the instruction to MOV PC, r, where r is
+the register used in the BX instruction.  See AAPCS32_ for
+details.  The symbol is unused and may even be the NULL symbol (index 0).
+
+R_ARM_TARGET1 is processed in a platform-specific manner. It may only be used
+in sections with the types SHT_INIT_ARRAY, SHT_PREINIT_ARRAY, and
+SHT_FINI_ARRAY.  The relocation must be processed either in the same way as
+R_ARM_REL32 or as R_ARM_ABS32: a virtual platform must specify which method is
+used.  If the relocation is processed as R_ARM_REL32 then the section may be
+marked read-only and coalesced with other read-only data, otherwise it may only
+be marked read-only if it does not require dynamic linking.
+
+R_ARM_TARGET2 is processed in a platform-specific manner. It is used to encode
+a data dependency that will only be dereferenced by code in the run-time
+support library.
+
+Proxy generating relocations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A number of relocations generate proxy locations that are then subject to
+dynamic relocation. The proxies are normally gathered together in a single
+table, called the Global Offset Table or GOT. The following table lists the
+relocations that generate proxy entries.
+
+.. table:: Proxy generating relocations
+  :widths: 10 27 63
```

Comment:
as described above on the version history table, please use a class to specify the width